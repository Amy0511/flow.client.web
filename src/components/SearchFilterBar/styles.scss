@import "~@boomerang/boomerang-styles/lib/colors";

.b-search-filter {
  display: flex;
}

.b-search-filter__search {
  flex: 1 1 auto;
  margin-right: 2rem;

  .bmrg--b-search-bar {
    background: $white;
  }

  .bmrg--b-search-bar__textbox.--bmrg-white {
    background: $white;
  }
}

.b-search-filter__filter {
  width: 10rem;

  .bx--list-box {
    background-color: $teal-20;
    border-radius: 0.25rem;
    box-shadow: none;
    cursor: pointer;
    height: 2.625rem;
    max-height: 2.625rem;

    &::before {
      content: url("~Assets/svg/configure.svg");
      position: absolute;
      height: 2rem;
      width: 2rem;
      right: 1rem;
      top: 0.2rem;
    }
  }

  .bx--list-box__field {
    appearance: none;
    &:focus {
      box-shadow: none;
    }
  }

  .bx--list-box__selection--multi {
    background-color: $gray-90;
    &:hover {
      background-color: $gray-90;
    }
  }

  .bx--list-box__menu-icon {
    display: none;
  }

  .bx--list-box__menu {
    min-width: 10rem;
    width: max-content;
    max-width: 15rem;
    background-color: $white;
    max-height: 7.5rem;
    right: 0;
    left: unset;
  }

  .bx--list-box__menu-item--highlighted {
    background: $cool-gray-1;
    text-decoration: none;
  }

  .bx--list-box__menu-item--active {
    .bx--checkbox-label::before {
      background-color: $teal-20 !important;
      border-color: $teal-20 !important;
    }
  }
}
