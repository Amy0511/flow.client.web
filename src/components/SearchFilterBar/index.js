import React, { Component } from "react";
import PropTypes from "prop-types";
import { MultiSelect } from "carbon-components-react";
import SearchBar from "@boomerang/boomerang-components/lib/SearchBar";
import "./styles.scss";

class SearchFilterBar extends Component {
  static propTypes = {
    teams: PropTypes.array.isRequired,
    handleSearchFilter: PropTypes.func.isRequired
  };

  state = {
    searchQuery: "",
    selectedItems: []
  };

  handleOnSearchInputChange = e => {
    const searchQuery = e.target.value;
    this.setState({ searchQuery }, () => {
      this.handleSearchFilter();
    });
  };

  handleOnSearchClear = () => {
    this.setState({ searchQuery: "" }, () => {
      this.handleSearchFilter();
    });
  };

  handleOnMultiSelectChange = e => {
    const selectedItems = e.selectedItems;
    this.setState({ selectedItems }, () => {
      this.handleSearchFilter();
    });
  };

  handleSearchFilter = () => {
    const { searchQuery, selectedItems } = this.state;

    this.props.handleSearchFilter(searchQuery, selectedItems);
  };

  formatTeams = teams => {
    return teams.map(team => ({ id: team.id, text: team.name }));
  };

  render() {
    const { teams } = this.props;

    return (
      <div className="b-search-filter">
        <div className="b-search-filter__search">
          <SearchBar
            theme="bmrg-white"
            onChange={this.handleOnSearchInputChange}
            onClear={this.handleOnSearchClear}
            value={this.state.searchQuery}
          />
        </div>
        <div className="b-search-filter__filter">
          <MultiSelect
            useTitleInItem={false}
            label="Filter"
            invalid={false}
            onChange={this.handleOnMultiSelectChange}
            items={teams.map(team => ({ id: team.id, text: team.name }))}
            itemToString={item => (item ? item.text : "")}
          />
        </div>
      </div>
    );
  }
}

export default SearchFilterBar;
