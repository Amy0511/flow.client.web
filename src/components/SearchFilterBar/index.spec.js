import React from "react";
import renderer from "react-test-renderer";
import SearchFilterBar from "./index";

const props = {
  options: [{ name: "test", workflows: [] }],
  handleSearchFilter: jest.fn(),
  debounceTimeout: 300,
  filterItems: [],
  multiselect: false,
  selectedOption: "test"
};

describe("SearchFilterBar --- Snapshot", () => {
  it("Capturing Snapshot of SearchFilterBar", () => {
    const renderedValue = renderer.create(<SearchFilterBar {...props} />);
    expect(renderedValue).toMatchSnapshot();
  });
});
