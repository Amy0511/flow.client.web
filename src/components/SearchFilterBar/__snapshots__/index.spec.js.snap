// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilterBar --- Snapshot Capturing Snapshot of SearchFilterBar 1`] = `
<div
  className="b-search-filter"
>
  <div
    className="b-search-filter__search"
  >
    <div
      className="bmrg--c-search-bar"
    >
      <div
        className="bmrg--b-search-bar --bmrg-white"
      >
        <svg
          className="bmrg--b-search-bar__icon --bmrg-white"
          height="20px"
          version="1.1"
          viewBox="0 0 18 20"
          width="18px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
            id="CI-Sam"
            stroke="none"
            strokeWidth="1"
          >
            <g
              id="CI---Landing---green-Copy-6"
              transform="translate(-31.000000, -170.000000)"
            >
              <g
                id="Group-11"
                transform="translate(30.000000, 170.000000)"
              >
                <g
                  id="search"
                >
                  <rect
                    height="20"
                    id="Rectangle-2"
                    width="20"
                    x="0"
                    y="0"
                  />
                  <path
                    d="M8.99971488,10.135742 C10.957325,12.1112984 14.1428734,12.111537 16.1007199,10.135742 C18.05833,8.16018553 18.05833,4.9451952 16.1007199,2.96963875 C14.1428734,0.993843756 10.957325,0.994082292 8.99971488,2.96963875 C7.04186838,4.94543374 7.04186838,8.15994699 8.99971488,10.135742 Z M16.8098276,10.8513505 C14.7576698,12.9223214 11.5813398,13.1832799 9.24530253,11.6354188 L1.42857143,19.5238095 L1.42857143,16.1843033 L7.57795384,9.97854665 C5.97655217,7.61250651 6.21173957,4.35195578 8.29060713,2.25403027 C10.6394084,-0.11630352 14.46079,-0.116542056 16.8098276,2.25403027 C19.1586289,4.62436407 19.1586289,8.48101666 16.8098276,10.8513505 Z"
                    fill="#40D5BB"
                    id="magnifying-glass"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
        <input
          className="bmrg--b-search-bar__textbox --bmrg-white"
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        
      </div>
    </div>
  </div>
  <div
    className="b-search-filter__filter"
  >
    <div
      className="bx--form-item"
    >
      <div
        className="bx--select"
      >
        <label
          className="bx--label bx--visually-hidden"
        >
          Select
        </label>
        <select
          className="bx--select-input"
          defaultValue="test"
          onChange={[Function]}
          useTitleInItem={false}
        >
          <option
            className="bx--select-option"
            disabled={false}
            hidden={false}
            value="none"
          >
            All
          </option>
          <optgroup
            className="bx--select-optgroup"
            disabled={false}
            label="test"
          />
        </select>
        <svg
          alt="open list of options"
          aria-label="open list of options"
          className="bx--select__arrow"
          fillRule="evenodd"
          height="5"
          role="img"
          viewBox="0 0 10 5"
          width="10"
        >
          <title>
            open list of options
          </title>
          <path
            d="M0 0l5 4.998L10 0z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
